src/libs/openReportActionComposeViewWhenClosingMessageEdit/index.native.js

```diff
@@ -1,8 +1,19 @@
 import {Keyboard} from 'react-native';
 import * as Composer from '../actions/Composer';
 
-export default () => {
-    const keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', () => {
+let keyboardDidHideListener = null;
+export default (shouldShow) => {
+    if (keyboardDidHideListener) {
+        keyboardDidHideListener.remove();
+        keyboardDidHideListener = null;
+    }
+
+    if (!shouldShow || !Keyboard.isVisible()) {
+        Composer.setShouldShowComposeInput(shouldShow);
+        return;
+    }
```

Comment:
```suggestion
    if (!shouldShow) {
        Composer.setShouldShowComposeInput(false);
        return;
    }
```

Isn't that sufficient?

Reply:
This code means 
- if `shouldSow` is false, hide composer immediately
- if `shouldShow` is true and the keyboard isn't visible, show composer immediately.

I think it's reasonable. What do you think?

Reply:
I don't see why we are bringing keyboard visibility logic here

Reply:
If keyboard was hidden, `keyboardDidHide` event never happens until we open keyboard and hide it. This means main compose won't be shown if we don't open and hide the keyboard. We should avoid this.

Reply:
Ah yes that would make sense but this logic for the other block of code and not the highlighted one.

```js
    if (!shouldShow) {
        Composer.setShouldShowComposeInput(false);
        return;
    }

    if (!Keyboard.isVisible()) {
        Composer.setShouldShowComposeInput(true);
        return;
    }
```


BTW, is there a reproducible bug for this case?

Reply:
Ah that code is actually the same as yours. I think it's better to decouple the logic into two conditions for readability and add a comment.

Reply:
Splitted logic and added comment